File: live-test/java/src/main/java/demo/Probes.java
package demo;

class Probe {
    int target() {
        return 42;
    }

    int sameClassCaller() {
        return target() + 1;
    }
}

class ProbeAux {
    static int freeProdCaller() {
        return new Probe().target();
    }
}

class ProbeProdChild extends Probe {
}

File: live-test/php/src/Probes.php
<?php
namespace Demo;

class Probe
{
    public function target(): int
    {
        return 42;
    }

    public function sameClassCaller(): int
    {
        return $this->target() + 1;
    }
}

function freeProdCaller(): int
{
    return (new Probe())->target();
}

class ProbeProdChild extends Probe
{
}
